backend/src/main/java/backend/service/impl/SolutionServiceImpl.java

```diff
@@ -106,4 +110,13 @@ public SolutionDtoWithId updateScore(int id, SolutionScoreDto solutionScoreDto)
         solutionRepository.save(solution);
         return SolutionConverter.getSolutionDtoWithId(solution);
     }
+
+    @Override
+    public void sendEmail(int id, String username) throws SolutionNotFoundException, UserNotFoundException {
+        UserTestSolution solution = solutionRepository.findById(id)
+                .orElseThrow(SolutionNotFoundException::new);
+        User user = userRepository.findByUsernameIgnoreCase(username)
+                .orElseThrow(UserNotFoundException::new);
```

Comment:
username jako parametr jest zbędne, użytkownik to pole obiektu solution

Reply:
Faktycznie